src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBuffer.cs

```diff
@@ -109,6 +113,9 @@ internal struct Storage
         private bool _isNull;
         private StorageType _type;
         private Storage _value;
+#if !NET8_0_OR_GREATER
+        private SqlGuid _sqlGuid;
+#endif
```

Comment:
This increases the size of the SqlBuffer type by guid+bool on all previous runtimes, this is almost doubling the size. Since SqlBuffer is used in every read this is going to cause a lot of additional memory traffic. I don't think this is a good idea.

Reply:
Yep, it is increases the size of SqlBuffer for platforms before .NET8, by 8 bytes. And that's it, no other downsides.

Reply:
That's a big downside. Every read goes through an SqlBuffer and this almost doubles the memory required to do that.

Reply:
For GC there will be no difference, I guess, because number of references has not changed (SqlGuid is a struct).

If it is not acceptable... Well, what is an alternative? Keep the old implementation with boxing for non .NET8 platforms?

Reply:
Larger is worse for throughput on the heap. I've put quite a lot of effort into shrinking various objects to maximize throughput. 

At this point i think it's worth reverting your other PR so we get a clean release (we've been waiting 6 months for one) then I'd like you to add your covering test cases alone and then we can investigate slowly.

Reply:
Fair point.
OK, lets revert #2306 , reopen #2300 and focus on

1. bugs in SqlBuffer first
2. removing of boxing for .NET8
3. removing of boxing for the rest platforms

Sounds good?

Reply:
Just to point out that an actual benchmark should easily show what's significant here and what isn't.